test/unit/build-request-file-object.test.js

```diff
@@ -61,7 +61,8 @@ describe('buildRequestFileObject', () => {
     });
 
     it('should handle file object with a readable stream as its `value`', async () => {
-      const fileStream = fs.createReadStream(filepath);
+      // Note add an on error handler to avoid unhandled error events
+      const fileStream = fs.createReadStream(filepath).on('error', () => {});
```

Comment:
Good call. Should we fail the test on error?

Reply:
I did wonder whether this implied that the helper functions should attach an error handler in the stream cases so that they could reject for any stream errors, but the behaviour seemed to be relying on resolving as `null`. As it currently stands the function caller (sdk/user?) would have to attach an error listener to the stream they create or risk this unhandled error event (or missing the error).

I wasn't really sure whether it made sense to suppress the error in the helper code instead of the test code (i.e. leaving the current behaviour of resolving as null, but avoiding any chance of an unhandled errors) rather than pushing the responsibility down. Ofc it is a little scope-creep as it isn't really related to the axios upgrade at all, just something that happened to be detected now that wasn't before.